Solutions/Marain.Workflow.Abstractions/Marain/Workflows/IWorkflowInstanceChangeLogWriter.cs

```diff
@@ -0,0 +1,26 @@
+﻿// <copyright file="IWorkflowInstanceChangeLogWriter.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Workflows
+{
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Defines the interface for writing log records describing the history of workflow instance changes.
+    /// </summary>
+    public interface IWorkflowInstanceChangeLogWriter
+    {
+        /// <summary>
+        /// Creates a workflow instance change record.
+        /// </summary>
+        /// <param name="trigger">The trigger that caused the update to the workflow instance, from which to create a snapshot; or null if this was an initialization operation for the workflow instance.</param>
+        /// <param name="instance">The workflow instance in its updated state after having processed the trigger, from which to create a snapshot.</param>
+        /// <param name="partitionKey">The partition key for the record.</param>
```

Comment:
The fact that only one of the three storage implementations actually uses this partition key - and has a fallback for when it's null - makes me question whether it should be the job of the consumer to specify it. Why not get rid of it from the interface and allow the implementation to use whatever the most appropriate partition key is?

If it's necessary to allow the consumer to control the partition key, could this be dealt with by allowing some kind of "partition key selection strategy" to be supplied to the Cosmos implementation of the interface?

Reply:
I agree. I followed the pattern already found in `IWorkflowEngine`. 

I've added a commit to remove the partition key from the public API.